org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ScalaProject.scala

```diff
@@ -427,20 +427,27 @@ class ScalaProject private (val underlying: IProject) extends ClasspathManagemen
   /** Compiler settings that are honored by the presentation compiler. */
   private def isPCSetting(settings: Settings): Set[Settings#Setting] = {
     import settings.{ plugin => pluginSetting, _ }
-    Set(deprecation,
+
+    val compilerPluginSettings: Set[Settings#Setting] = Set(pluginOptions,
+      pluginSetting,
+      pluginsDir)
+
+    val generalSettings: Set[Settings#Setting] = Set(deprecation,
       unchecked,
-      pluginOptions,
       verbose,
       Xexperimental,
       future,
       Ylogcp,
-      pluginSetting,
-      pluginsDir,
       YpresentationDebug,
       YpresentationVerbose,
       YpresentationLog,
       YpresentationReplay,
       YpresentationDelay)
+
+    if (effectiveScalaInstallation().version == ScalaInstallation.platformInstallation.version)
+      generalSettings ++ compilerPluginSettings
+    else
+      generalSettings
```

Comment:
You're saying that the presentation compiler error markers are gone... so that's what should happen. In general, a compiler plugin that doesn't match the exact Scala compiler version can't be instantiated, so the presentation compiler will simply ignore it.

If the compiler plugin doesn't match the installation, the earlier check in `checkClasspath` will flag it as an error marker at *build time*. For building we can't skip compiler plugins, they are required for a correct output. So the user will have to set a correct Scala installation. This would fix the build. The presentation compiler will continue to skip the compiler plugin, since the PC is locked down on the platform installation.

The default path, `misc/scala-devel/plugins` is probably wrong, I don't know exactly where it comes from. Probably the Scala compiler itself.